launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -107,6 +121,30 @@ public static void setConfig(String name, String value) {
     launcherConfig.put(name, value);
   }
 
+  /**
+   * Specifies that Spark Application be stopped if current process goes away.
+   * It tries stop/kill Spark Application if {@link LauncherServer} goes away.
+   *
+   * @since 2.2.0
+   * @return This launcher.
+   */
+  public SparkLauncher stopIfLauncherShutdown() {
```

Comment:
Follow up: I can see why someone would want this in "child proc" mode, but they can achieve that today by installing their own shutdown hook in the launcher side. It's not as guaranteed to work as handling it on the app side when the launcher disconnects, but you need to be able to solve the "stopOnShutdown" vs. "disconnect" behavior before adding this API.